lib/events/kafka_listener.rb

```diff
@@ -39,8 +39,8 @@ def raw_process(event)
         if event.payload.class == Hash && event.payload.has_key?("params") 
           # check availability from source doesn't have headers
           insights_headers['x-rh-insights-request-id'] = "unknown"
-          insights_headers['x-rh-identity'] = identity_from_external_tenant(event.payload["params"]["external_tenant"])
-        else 
+          insights_headers['x-rh-identity'] = Headers::Service.x_rh_identity_tenant_user(event.payload["params"]["external_tenant"])
```

Comment:
@hsong-rh Can we check if the key external_tenant exists, we are only checking for "params"